test/Otoken.test.ts

```diff
@@ -0,0 +1,125 @@
+import {OtokenInstance, MockERC20Instance} from '../build/types/truffle-types'
+import BigNumber from 'bignumber.js'
+
+const {expectRevert} = require('@openzeppelin/test-helpers')
+
+const Otoken = artifacts.require('Otoken.sol')
+const MockERC20 = artifacts.require('MockERC20.sol')
+const ZERO_ADDR = '0x0000000000000000000000000000000000000000'
+const ETH_ADDR = ZERO_ADDR
+
+contract('Otoken', ([deployer, mockAddressBook, random]) => {
+  let otoken: OtokenInstance
+  let usdc: MockERC20Instance
+
+  // let expiry: number;
+  const strikePrice = new BigNumber(200).times(new BigNumber(10).exponentiatedBy(18))
+  const expiry = 1601020800 // 2020/09/25 0800 UTC
+  const isPut = true
+
+  before('Deployment', async () => {
+    // Need another mock contract for addressbook when we add ERC20 operations.
+    otoken = await Otoken.new(mockAddressBook)
+    usdc = await MockERC20.new('USDC', 'USDC')
+  })
+
+  describe('Otoken Initialization', () => {
+    it('should be able to initialize with put parameter correctly', async () => {
+      await otoken.init(ETH_ADDR, usdc.address, usdc.address, strikePrice, expiry, isPut, {from: deployer})
```

Comment:
shouldn't this only be callable from Factory?

Reply:
we don't need to restrict that in my opinion, we only need to make sure it can only be called once, because our factory will call init once right after the creation, and we can just use the `Initializer` modifier from OZ

Reply:
If we really want to reduce complexity, we definitely don't need that one more modifier. 

The onlyFactory feature is automatically satisfied when you combine the Factory and the Otoken, so that should be included in the integration test: Factory ensure it calls init right after deploy the contract, and Otoken make sure the function can only be called once.

So in the unit test for Otoken, we only need to test that it can only be called once.